react/props/index.js

```diff
@@ -0,0 +1,157 @@
+import PropTypes from 'prop-types'
+
+import productPropTypes from 'vtex.product-details/propTypes'
```

Comment:
🤔 its a trap!

Reply:
Both components receives from the `ProductQuery` the same shape of `Product` since they share the `ProductContextProvider`. I understand that this solution have some risks (the `propTypes` of `ProductDetails` could change in any moment), but why we do not create a poll of the most common `propTypes` shapes and make the apps use these `propTypes`? I think that this should be a better solution than rewrite great part of the validation for every component that receives the same Query from the context.